assets/src/edit-story/app/media/mediaProvider.js

```diff
@@ -31,45 +31,67 @@ import Context from './context';
 function MediaProvider({ children }) {
   const { state, actions } = useMediaReducer();
   const { uploadVideoFrame } = useUploadVideoFrame();
-  const { mediaType, searchTerm } = state;
+  const { page, mediaType, searchTerm } = state;
   const {
     fetchMediaStart,
     fetchMediaSuccess,
     fetchMediaError,
     resetFilters,
     setMediaType,
     setSearchTerm,
+    setPage,
   } = actions;
 
   const {
     actions: { getMedia },
   } = useAPI();
 
-  const fetchMedia = useCallback(() => {
-    fetchMediaStart();
-    getMedia({ mediaType, searchTerm })
-      .then((media) => {
-        fetchMediaSuccess({ media, mediaType, searchTerm });
-      })
-      .catch(fetchMediaError);
-  }, [
-    fetchMediaError,
-    fetchMediaStart,
-    fetchMediaSuccess,
-    getMedia,
-    mediaType,
-    searchTerm,
-  ]);
+  const fetchMedia = useCallback(
+    ({ page: p = 1 } = {}) => {
+      fetchMediaStart({ page: p });
+      getMedia({ mediaType, searchTerm, page: p })
+        .then(({ data, headers }) => {
+          const totalPages = parseInt(headers.get('X-WP-TotalPages'));
```

Comment:
For my info: is this actually WP approach to use response headers for pagination data?

Reply:
[Yep...](https://developer.wordpress.org/rest-api/using-the-rest-api/pagination/)